### DISCUSSION TITLE
Build Time per Deployment

### DISCUSSION THREAD

**Discussion by FutureBuddha** [vercel/community#5232] [Help]
### Summary

I wish you a nice day,

I am planning a big project with Astro and intend to incorporate a lot of data into static sites. The issue I am facing is that I expect the build time to be longer than 45 minutes. At this point, I am not certain what the actual build time will be, but it is likely to be more than two hour.

What would be the cost to decrease the Build Time per Deployment, and is it even possible?

Thank you very much in advance and best regards.

### Steps to Reproduce

Just try to call a large amount of data via API (especially images) and use Astro to build a static site for each set of data

**Comment by amyegan** [comment]
Hey @FutureBuddha! There are a couple of options for this. You could use [on-demand rendering](https://docs.astro.build/en/guides/server-side-rendering/) to reduce the amount of pages that need to be generated during the build step. This could dramatically reduce build time depending on your setup.

Or you could use Vercel CLI to [deploy from a local build (prebuilt)](https://vercel.com/docs/cli/deploying-from-cli#deploying-from-local-build-prebuilt) and bypass the 45 minute build limit. If you have a lot of files do deploy, you may also need to use the [`--archive=tgz` option](https://vercel.com/docs/cli/deploy#archive) to compress the code into a single file before uploading it.

The Astro folks may have some more specific recommendations about configuring the framework, so it's probably also worth asking for their advice over on Discord: https://astro.build/chat
